admin/listPurchases/listView.html

```diff
@@ -183,11 +176,29 @@
     e.preventDefault();
     entryPageIndex = inputEle.value - 1;
     updateContent();
-    generateSubPageNavigation();
 })
 
+const orderSearchBar = document.getElementById("order-search-bar");
+
+orderSearchBar.addEventListener("input", () =>{entryPageIndex = 0; updateContent();});
+
 function updateContent(){
-    renderEntriesToList(entries.filter(currentOption.filter).slice(entryPageIndex * 10, entryPageIndex * 10 + 10));
+    if(!orderSearchBar.value){
+        renderEntriesToList(entries.filter(currentOption.filter).slice(entryPageIndex * 10, entryPageIndex * 10 + 10));
+    }else{
+        renderEntriesToList(entries.filter(currentOption.filter).filter(searchBarFilter).slice(entryPageIndex * 10, entryPageIndex * 10 + 10));
```

Comment:
These lines are way too long and actually the if and else case share a lot of logic - but that's hard to read from this! Instead, do fewer things in each step to make the code more readable.

Think of it this way: you want to make it clear to readers exactly what differs in the two cases - and exactly what is the same. Which means making the if-else statement contain as little code as possible in each branch, while keeping most of the function's code before or after the if-else